chain/chain/src/resharding/manager.rs

```diff
@@ -146,16 +147,19 @@ impl ReshardingManager {
     }
 
     /// Store in the database the mapping of ShardUId from children to the parent shard,
-    /// so that subsequent accesses to the State will use the parent shard's UId as a prefix for the database key.
+    /// so that subsequent accesses to the State will use the parent shard's UId prefix
+    /// as a prefix for the database key.
+    // TODO(resharding) add testloop where grandparent ShardUId is used
     fn set_state_shard_uid_mapping(
         &mut self,
         split_shard_event: &ReshardingSplitShardParams,
     ) -> io::Result<()> {
         let mut store_update = self.store.trie_store().store_update();
         let parent_shard_uid = split_shard_event.parent_shard;
+        let parent_shard_uid_prefix = get_shard_uid_mapping(&self.store, parent_shard_uid);
```

Comment:
If we do this in a loop until the last iteration returns the same value as the previous (meaning we navigated all children), would it be a good way to support any number of sub reshardings?

Although I don't see it likely going further down than shard grand children..

Reply:
I think it already works for any number of sub reshardings. E.g. split A -> B -> C, then we save to the db mapping from C to A. If we then split C -> D, then we would save to the db mapping from D to get_mapping(C)=A.
Is this what you meant?

Reply:
Yes.. I think it should work as you said. We'll be sure after adding tests :)